tests/run_test.go

```diff
@@ -0,0 +1,217 @@
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package tests
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestNormal(t *testing.T) {
+	pebbleDir := t.TempDir()
+
+	layerYAML := `
+services:
+    svc1:
+        override: replace
+        command: /bin/sh -c "{{.svc1Cmd}}"
+        startup: enabled
+    svc2:
+        override: replace
+        command: /bin/sh -c "{{.svc2Cmd}}"
+        startup: enabled
+`
+	svc1Cmd := fmt.Sprintf("touch %s; sleep 1000", filepath.Join(pebbleDir, "svc1"))
+	svc2Cmd := fmt.Sprintf("touch %s; sleep 1000", filepath.Join(pebbleDir, "svc2"))
+	layerYAML = strings.Replace(layerYAML, "{{.svc1Cmd}}", svc1Cmd, -1)
+	layerYAML = strings.Replace(layerYAML, "{{.svc2Cmd}}", svc2Cmd, -1)
+
+	createLayer(t, pebbleDir, "001-simple-layer.yaml", layerYAML)
+
+	_ = pebbleRun(t, pebbleDir)
+
+	expectedServices := []string{"svc1", "svc2"}
+	waitForServices(t, pebbleDir, expectedServices, time.Second*3)
+}
+
+func TestCreateDirs(t *testing.T) {
+	tmpDir := t.TempDir()
+	pebbleDir := filepath.Join(tmpDir, "PEBBLE_HOME")
+
+	logsCh := pebbleRun(t, pebbleDir, "--create-dirs")
+	expectedLogs := []string{"Started daemon"}
+	if err := waitForLogs(logsCh, expectedLogs, time.Second*3); err != nil {
+		t.Errorf("Error waiting for logs: %v", err)
+	}
+
+	if _, err := os.Stat(pebbleDir); err != nil {
+		t.Errorf("pebble run --create-dirs failed: %v", err)
+	}
+}
+
+func TestHold(t *testing.T) {
+	pebbleDir := t.TempDir()
+
+	layerYAML := `
+services:
+    svc1:
+        override: replace
+        command: /bin/sh -c "{{.svc1Cmd}}"
+        startup: enabled
+`
+	svc1Cmd := fmt.Sprintf("touch %s ; sleep 1000", filepath.Join(pebbleDir, "svc1"))
+	layerYAML = strings.Replace(layerYAML, "{{.svc1Cmd}}", svc1Cmd, -1)
+
+	createLayer(t, pebbleDir, "001-simple-layer.yaml", layerYAML)
+
+	logsCh := pebbleRun(t, pebbleDir, "--hold")
+	expectedLogs := []string{"Started daemon"}
+	if err := waitForLogs(logsCh, expectedLogs, time.Second*3); err != nil {
+		t.Errorf("Error waiting for logs: %v", err)
+	}
+
+	// Sleep a second before checking services because immediate check
+	// can't guarantee that svc1 is not started shortly after the log "Started daemon".
+	time.Sleep(time.Second)
+
+	_, err := os.Stat(filepath.Join(pebbleDir, "svc1"))
+	if err == nil {
+		t.Error("pebble run --hold failed, services are still started")
+	} else {
+		if !os.IsNotExist(err) {
+			t.Errorf("Error checking service %s: %v", "svc1", err)
+			fmt.Printf("Error checking the file: %v\n", err)
+		}
+	}
+}
+
+func TestHttpPort(t *testing.T) {
+	pebbleDir := t.TempDir()
+
+	port := "4000"
+	logsCh := pebbleRun(t, pebbleDir, "--http=:"+port)
+	expectedLogs := []string{"Started daemon"}
+	if err := waitForLogs(logsCh, expectedLogs, time.Second*3); err != nil {
+		t.Errorf("Error waiting for logs: %v", err)
+	}
+
+	if !isPortUsedByProcess(t, port, "pebble") {
```

Comment:
Once again, I think this helper obscures the test (I had no idea it would use `lsof`, for example). Also, I think we should just make an `http.Get` call to one of the HTTP endpoints (probably `http://localhost:<port>/v1/health`).

```go
resp, err := http.Get("http://localhost:4000/v1/health")
if err != nil { /* error */ }
defer resp.Body.Close()
if resp.StatusCode != 200 { /* error */ }
```